src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,107 @@
+"""
+Passkey Tests.
+
+:requirement: passkey
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.Client)
+def test_passkey__register__sssctl(client: Client, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a key with sssctl
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key.
+        2. Check the output.
+    :expectedresults:
+        1. New key is registered
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = client.sssctl.passkey_register(
+        username="user1",
+        domain="ldap.test",
+        pin=123456,
+        device=f"{moduledatadir}/umockdev.device",
+        ioctl=f"{moduledatadir}/umockdev.ioctl",
+        script=f"{testdatadir}/umockdev.script",
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with sssctl"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a passkey with the IPA command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use ipa command to register a FIDO2 key.
+        2. Check the output that contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered with IPA command.
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = (
+        ipa.user("user1")
+        .add()
+        .passkey_add_register(
+            pin=123456,
+            device=f"{moduledatadir}/umockdev.device",
+            ioctl=f"{moduledatadir}/umockdev.ioctl",
+            script=f"{testdatadir}/umockdev.script",
+        )
+    )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip(), "Failed to register a key with the IPA command"
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.LDAP)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.AD)
+@pytest.mark.topology(KnownTopology.Samba)
+def test_passkey__su(client: Client, provider: GenericProvider, moduledatadir: str, testdatadir: str):
```

Comment:
Add the following marker to this test case:
```
@pytest.mark.builtwith(client="passkey", ipa="passkey")
```

Reply:
@pytest.mark.builtwith(client="passkey", ipa="passkey"), here is  ipa="passkey is for ipa topology?

Reply:
Yes, it's for the IPA topology. You can add it after all the topology markers, and before the function definition.

Reply:
Hmm, I think we should add topology information to the builtwith marker as well otherwise the test will not run even on non-ipa topologies.

Reply:
Somethink like:

```python
@pytest.mark.builtwith(client="passkey")
@pytest.mark.builtwith(KnownTopology.IPA, ipa="passkey")
```

Reply:
Okay but here we have other topology also, cant we add them?
AD or LDAP?

Reply:
> Hmm, I think we should add topology information to the builtwith marker as well otherwise the test will not run even on non-ipa topologies.

Strictly speaking this isn't necessary for the passkey feature. We only need to test fedora39+, as these are the only versions where the feature is enabled. With the current proposal this is already covered.

> Okay but here we have other topology also, cant we add them? AD or LDAP?

No need for that. We need a certain version of IPA installed to test the feature. In contrast, it can be tested with any version of AD and LDAP so we don't need to filter them.

Reply:
Okay, got it.
Thanks!

Reply:
updated.

Reply:
> > Hmm, I think we should add topology information to the builtwith marker as well otherwise the test will not run even on non-ipa topologies.
> 
> Strictly speaking this isn't necessary for the passkey feature. We only need to test fedora39+, as these are the only versions where the feature is enabled. With the current proposal this is already covered.

Why it isn't needed?

These tests are part of sssd release tarball, anyone can run them against custom configuration of SSSD. So anyone can built SSSD `--without-passkey` and still run the test. This by the way happens on RHEL8 - same SSSD version as on RHEL9 with the same test suite just built without passkey support.

> > Okay but here we have other topology also, cant we add them? AD or LDAP?
> 
> No need for that. We need a certain version of IPA installed to test the feature. In contrast, it can be tested with any version of AD and LDAP so we don't need to filter them.

Additionally, for plain LDAP, we should check the presence of passkey schema in order to run the tests.